tests/unit/ui/components/TestShowMoreCard.js

```diff
@@ -1,105 +1,223 @@
+    root.instance().truncateToMaxHeight({ clientHeight: MAX_HEIGHT + 1 });
 
-    expect(root).not.toHaveClassName('ShowMoreCard--expanded');
+    applyUIStateChanges({ root, store });
 
-    // We have to shallow this prop because we pass a component to it.
-    expect(shallow(root.prop('footerLink')).html()).toContain(
-      '<span class="visually-hidden">Expand to</span> Read more',
-    );
+    expect(root).toHaveProp('footerLink');
+    const footerLink = root.prop('footerLink');
+    const cardLink = shallow(footerLink);
+    const moreLink = cardLink.find('.ShowMoreCard-expand-link');
 
-    // We cannot directly target the footer link, so we simulate the click here.
-    root.instance().onClick(createFakeEvent());
-    root.update();
+    moreLink.simulate('click', createFakeEvent());
 
-    expect(root).toHaveClassName('ShowMoreCard--expanded');
+    applyUIStateChanges({ root, store });
 
-    expect(root).toHaveProp('footerLink', null);
+    expect(root).toHaveClassName('ShowMoreCard--expanded');
   });
 
   it('is expanded by default', () => {
-    const root = render({ children: 'Hello I am description' });
+    const root = render();
 
     expect(root).toHaveClassName('ShowMoreCard--expanded');
   });
 
   it('truncates the contents if they are too long', () => {
-    const root = render({ children: 'Hello I am description' });
+    const { store } = dispatchClientMetadata();
+    const root = render({ store });
 
+    // We are simulating the truncate method call.
     root.instance().truncateToMaxHeight({ clientHeight: MAX_HEIGHT + 1 });
-    root.update();
+
+    applyUIStateChanges({ root, store });
 
     expect(root).not.toHaveClassName('ShowMoreCard--expanded');
   });
 
   it('renders className', () => {
     const className = 'test';
-    const root = render({
-      children: <p>Hi</p>,
-      className,
-    });
-
+    const root = render({ className });
     expect(root).toHaveClassName(className);
   });
 
   it('renders children', () => {
     const root = render({ children: 'Hello I am description' });
+    const contents = root.find('.ShowMoreCard-contents');
+    expect(contents).toHaveText('Hello I am description');
+  });
+
+  it('dispatches if the html children has changed', () => {
+    const { store } = dispatchClientMetadata();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    /* eslint-disable react/no-danger */
+    const root = render({
+      store,
+      children: (
+        <div
+          dangerouslySetInnerHTML={{
+            __html: '<span>First component text.</span>',
+          }}
+        />
+      ),
+    });
+
+    root.setProps({
+      children: (
+        <div
+          dangerouslySetInnerHTML={{
+            __html: '<span>This is different Text.</span>',
+          }}
+        />
+      ),
+    });
+    /* eslint-enable react/no-danger */
 
-    expect(root.childAt(0)).toHaveText('Hello I am description');
+    sinon.assert.called(dispatchSpy);
```

Comment:
The component does not call `dispatch` itself directly. I assume this is called when `setUIState` is called internally, but it seems like it would be better to just check for `setUIState` to be called, rather than checking for `dispatch` to be called.

Reply:
I see what you mean but for some reason when I spy on setUIState, the tests don't work. I'll revisit this tho to be sure..